test/k8s/service_helpers.go

```diff
@@ -865,13 +867,17 @@ func testExternalTrafficPolicyLocal(kubectl *helpers.Kubectl, ni *helpers.NodesI
 			testCurlFromOutside(kubectl, ni, tftpURL, count, true)
 		}
 
-		// Local requests should be load-balanced on kube-proxy 1.15+.
-		// See kubernetes/kubernetes#77523 for the PR which introduced this
-		// behavior on the iptables-backend for kube-proxy.
 		httpURL = getHTTPLink(node.node1IP, data.Spec.Ports[0].NodePort)
 		tftpURL = getTFTPLink(node.node1IP, data.Spec.Ports[1].NodePort)
-		testCurlFromPodInHostNetNS(kubectl, httpURL, count, 0, ni.K8s1NodeName)
-		testCurlFromPodInHostNetNS(kubectl, tftpURL, count, 0, ni.K8s1NodeName)
+
+		// Until https://github.com/cilium/cilium/issues/23481 has been fixed
+		if !ipsec {
```

Comment:
This should ideally be in a separate commit so we can (a) store the rationale and (b) just revert it later when we fix #23481.

Reply:
It would also help to document why specific callers of this function pass true or false. At a glance I couldn't quite establish why some sibling statements below pass ipsec=true or ipsec=false.

Reply:
My reasoning https://github.com/cilium/cilium/pull/23343#discussion_r1095662225 applies here too :-)

I hope we will add the externalTrafficPolicy=Local test cases to the `ci-datapath` sooner than #23481 gets fixed.

Do you still want me to move this to a separate commit?

Reply:
Not splitting the changes into separate commits mainly has the impacts of making the reviewer job harder, which is why I always encourage more effort in that regard. I see what you're saying about dead code, and I agree that it's better to make cleaner contributions to the code that isn't effectively deprecated. In this case, my last question above I don't understand the answer, which is why I had requested the rationale etc. above.

In order to make this more clear, let me ask the question on the relevant code rather than raising it in the reused function here.

Reply:
OK, looking at the later code I realize my confusion was that I thought the callers were not consistently passing the `ipsec` parameter, but they are actually consistent.

Reply:
Summary: Sure, you don't have to split the commits, but be aware that starting with not splitting the commits meant that I delayed review a lot more because I was more confused about what's going on. At this point it's sunk cost :ship:

Reply:
Blocking on this. I'm sorry because I understand you want to move on with these changes, but we really need to understand the impact for IPsec coverage here and that should be explained in the commit history. Deprecating ginkgo (where is that documented?) cannot be an excuse to have a poor commit history.